tests/unit_tests/pal_temperature/src/main.c

```diff
@@ -1,32 +1,200 @@
+	sid_pal_temperature_init();
+	strcpy(DT_N_NODELABEL_temp_FULL_NAME, "dummy_sensor");
+
+	RESET_FAKE(dummy_sensor_sample_fetch);
+	RESET_FAKE(dummy_sensor_channel_get);
 }
 
-/* It is required to be added to each test. That is because unity is using
- * different main signature (returns int) and zephyr expects main which does
- * not return value.
- */
-extern int unity_main(void);
+ZTEST_SUITE(temperature, NULL, NULL, setUp, NULL, NULL);
+
+ZTEST(temperature, test_sid_pal_temperature_init)
+{
+	sid_error_t error = sid_pal_temperature_init();
+	zassert_equal(SID_ERROR_NONE, error, "Initialization failed with error %d, expected %d.",
+		      error, SID_ERROR_NONE);
+}
+
+ZTEST(temperature, test_sid_pal_temperature_init_wrong_device)
+{
+	memset(DT_N_NODELABEL_temp_FULL_NAME, 0, sizeof(DT_N_NODELABEL_temp_FULL_NAME));
+	sid_error_t error = sid_pal_temperature_init();
+	zassert_equal(SID_ERROR_NOSUPPORT, error,
+		      "Initialization failed with error %d, expected %d.", error,
+		      SID_ERROR_NOSUPPORT);
+}
+
+ZTEST(temperature, test_get_value)
+{
+	dummy_sensor_sample_fetch_fake.return_val = 0;
+
+	dummy_sensor_channel_get_fake.custom_fake = custom_fake_dummy_sensor_channel_get;
+	dummy_sensor_channel_get_fake.return_val = 0;
+	custom_fake_dummy_sensor_channel_get_sensor_value = SENSOR_VALUE_FLOAT(15.123);
+
+	sid_error_t error = sid_pal_temperature_init();
+	zassert_equal(SID_ERROR_NONE, error, "Initialization failed with error %d, expected %d.",
+		      error, SID_ERROR_NONE);
+	int16_t temperature_C = sid_pal_temperature_get();
+	zassert_equal(15, temperature_C, "Unexpected temperature read %d, expected %d.",
+		      temperature_C, 15);
+
+	zassert_equal(1, dummy_sensor_sample_fetch_fake.call_count,
+		      "dummy_sensor_sample_fetch called %d times, expected %d",
+		      dummy_sensor_sample_fetch_fake.call_count, 1);
+	zassert_equal(1, dummy_sensor_channel_get_fake.call_count,
+		      "dummy_sensor_channel_get called %d times, expected %d",
+		      dummy_sensor_channel_get_fake.call_count, 1);
+}
+
+ZTEST(temperature, test_get_value_uninitialized)
+{
+	memset(DT_N_NODELABEL_temp_FULL_NAME, 0, sizeof(DT_N_NODELABEL_temp_FULL_NAME));
+
+	sid_error_t error = sid_pal_temperature_init();
+	zassert_equal(SID_ERROR_NOSUPPORT, error,
+		      "Initialization failed with error %d, expected %d.", error,
+		      SID_ERROR_NOSUPPORT);
+	int16_t temperature_C = sid_pal_temperature_get();
+	zassert_equal(INT16_MIN, temperature_C, "Unexpected temperature read %d, expected %d.",
+		      temperature_C, INT16_MIN);
+
+	zassert_equal(0, dummy_sensor_sample_fetch_fake.call_count,
+		      "dummy_sensor_sample_fetch called %d times, expected %d",
+		      dummy_sensor_sample_fetch_fake.call_count, 0);
+	zassert_equal(0, dummy_sensor_channel_get_fake.call_count,
+		      "dummy_sensor_channel_get called %d times, expected %d",
+		      dummy_sensor_channel_get_fake.call_count, 0);
+}
 
-int main(void)
+ZTEST(temperature, test_get_value_fetch_fail_1)
```

Comment:
Why to add `_1` at the end of test name?

```suggestion
ZTEST(temperature, test_get_value_fetch_fail)
```

Reply:
because sensor_sample_fetch_fake returns 1  (invalid value).

Next test returns -1, and test name ends with `negative_1`